model_compression_toolkit/keras/quantizer/mixed_precision/selective_quantize_config.py

```diff
@@ -0,0 +1,217 @@
+    from tensorflow.python.keras.layers import Layer
+else:
+    from keras.engine.base_layer import Layer
+from tensorflow.python.training.tracking.data_structures import ListWrapper
+from tensorflow_model_optimization.python.core.quantization.keras.quantize_config import QuantizeConfig
+
+from model_compression_toolkit.keras.quantizer.mixed_precision.selective_weights_quantizer import SelectiveWeightsQuantizer
+
+
+class SelectiveQuantizeConfig(QuantizeConfig):
+    """
+    SelectiveQuantizeConfig to use as a QuantizeCong for layers that are wrapped
+    for MP models. SelectiveQuantizeConfig holds a SelectiveWeightsQuantizer and uses
+    it to use quantized weight from a set of quantized weights (each one of the
+    quantized weights was quantized with different bitwidth).
+    At any given time, the SelectiveQuantizeConfig uses only one quantized weight
+    according to an "active" index - the index of a candidate weight quantization configuration
+    from a list of candidates that was passed to the SelectiveQuantizeConfig when it was initialized.
+    The "active" index can be configured as part of the SelectiveQuantizeConfig's API,
+    so a different quantized weight can be used in another time.
+    """
+
+    def __init__(self,
+                 node_q_cfg: List[CandidateNodeQuantizationConfig],
+                 float_weights: List[np.ndarray] = None,
+                 weight_attrs: List[str] = None):
+        """
+        Init a SelectiveQuantizeConfig instance.
+
+        Args:
+            weight_attrs: Attributes of the layer's weights to quantize, the
+            SelectiveQuantizeConfig is attached to.
+            float_weights: Float weights of the layer, the SelectiveQuantizeConfig is attached to.
+            node_q_cfg: Candidates quantization config the node has (the node from which
+            we built the layer that is attached to SelectiveQuantizeConfig).
+        """
+        # Make sure the candidates configurations arrived in a descending order.
+        curmax = (np.inf, np.inf)
+        n_candidate_bits = [(x.weights_quantization_cfg.weights_n_bits, x.activation_quantization_cfg.activation_n_bits)
+                            for x in node_q_cfg]
+        for candidate_bits in n_candidate_bits:
+            assert candidate_bits < curmax
+            curmax = candidate_bits
+
+        self.weight_attrs = weight_attrs
+        self.float_weights = float_weights
+
+        assert len(node_q_cfg) > 0, 'SelectiveQuantizeConfig has to receive' \
+                                            'at least one weight quantization configuration'
+        assert (not weight_attrs and not float_weights) or len(weight_attrs) == len(float_weights)
+
+        self.node_q_cfg = node_q_cfg
+        self.enable_weights_quantization = node_q_cfg[0].weights_quantization_cfg.enable_weights_quantization
+        self.enable_activation_quantization = node_q_cfg[0].activation_quantization_cfg.enable_activation_quantization
+
```

Comment:
Why do we take the first candidate flag?

Reply:
Adding assert that all candidates have same value for flag.
For later task - consider allowing some of the candidates to have disabled weights/activation quantization.